// proto/tts.proto
syntax = "proto3";
package tts;

service TTSService {
  rpc Synthesize(TTSRequest) returns (TTSResponse);
}

message TTSRequest {
  string text   = 1;
  string voice  = 2;
  string format = 3;
}

message VisemeEvent {
  int32 id      = 1;
  int32 time_ms = 2;
}

message TTSResponse {
  bytes audio_data          = 1;
  string format             = 2;
  repeated VisemeEvent visemes = 3;
}
